src/Octoshift/Services/ArchiveUploader.cs

```diff
@@ -160,4 +166,22 @@ private Uri GetNextUrl(IEnumerable<KeyValuePair<string, IEnumerable<string>>> he
         }
         throw new OctoshiftCliException("Location header is missing in the response, unable to retrieve next URL for multipart upload.");
     }
+
+    private void SetStreamSizeLimitFromEnvironment()
+    {
+        var envValue = _environmentVariableProvider.GithubOwnedStorageMultipartBytes();
+        if (!int.TryParse(envValue, out var limit) || limit <= 0)
+        {
+            return;
+        }
+
+        if (limit < MIN_MULTIPART_BYTES)
+        {
+            _log.LogWarning($"GITHUB_OWNED_STORAGE_MULTIPART_BYTES is set to {limit} bytes, but the minimum value is {MIN_MULTIPART_BYTES} bytes. Using default value of {_streamSizeLimit} bytes.");
+            return;
+        }
+
+        _streamSizeLimit = limit;
+        _log.LogInformation($"Stream size limit set to {_streamSizeLimit} bytes.");
```

Comment:
"Stream size" might be a little confusing.  How about using explicit language about what we're setting?


```suggestion
        _log.LogInformation($"Multipart upload part size set to {_streamSizeLimit} bytes.");
```